### PR TITLE
Light entity match SDF boolean for UserCommands.

### PR DESCRIPTION
**Pull request by bperseghetti**
# 🦟 Bug fix

## Summary
Unable to set multiple matching light entities like what is done in MaterialColor.
Adding an SDF boolean element `<set_all_light_entities>` to usercommands plugin allows for controlling matching light entities.

Waiting first on merge of https://github.com/gazebosim/gz-sim/pull/2286 for MaterialColor change.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Updated documentation (as needed)
- [x] Updated migration guide (as needed)
- [x] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by bperseghetti** [comment]
Will rebase this soon as https://github.com/gazebosim/gz-sim/pull/2286 gets merged in.

**Comment by bperseghetti** [comment]
Test just added, should have full coverage now.

**Comment by bperseghetti** [comment]
@osrf-jenkins run tests please

File: src/systems/user_commands/UserCommands.cc [hunk 1]
```diff
@@ -545,6 +548,7 @@ class gz::sim::systems::UserCommandsPrivate
 
   /// \brief Mutex to protect pending queue.
   public: std::mutex pendingMutex;
+
```
Comment (ahcorde) [inline src/systems/user_commands/UserCommands.cc:551]: remove

File: test/integration/user_commands.cc [hunk 2]
```diff
@@ -1031,22 +1031,350 @@ TEST_F(UserCommandsTest, GZ_UTILS_TEST_ENABLED_ONLY_ON_LINUX(Light))
...
+  EXPECT_EQ(nullptr, ecm);
+  server.Run(true, 1, false);
+  EXPECT_NE(nullptr, ecm);
+
+  msgs::Light req;
+  msgs::Boolean res;
+  transport::Node node;
+  bool result;
+  unsigned int timeout = 1000;
+  std::string service{"/world/lights_command/light_config"};
```
Comment (ahcorde) [inline test/integration/user_commands.cc:1096]: include `<string>`

Comment (bperseghetti) [reply test/integration/user_commands.cc:1096]: Hmmmm, not sure what you are suggesting with this one? I'm following how it is done in all other tests?:

https://github.com/gazebosim/gz-sim/blob/4f6bad50ad5505010ffa2e39330d661a914826e7/test/integration/user_commands.cc#L141

https://github.com/gazebosim/gz-sim/blob/4f6bad50ad5505010ffa2e39330d661a914826e7/test/integration/user_commands.cc#L814

https://github.com/gazebosim/gz-sim/blob/4f6bad50ad5505010ffa2e39330d661a914826e7/test/integration/user_commands.cc#L1698

etc...

Could you clarify this more?

File: src/systems/user_commands/UserCommands.cc [hunk 3]
```diff
@@ -680,6 +683,15 @@ void UserCommands::Configure(const Entity &_entity,
   this->dataPtr->node.Subscribe(lightTopic, &UserCommandsPrivate::OnCmdLight,
                                 this->dataPtr.get());
 
+  if (_sdf->HasElement("set_all_light_entities"))
+  {
+    this->dataPtr->iface->setAllLightEntities =
+      _sdf->Get<bool>("set_all_light_entities");
+    gzmsg << "Set all light entities: "
```
Comment (mjcarroll) [inline src/systems/user_commands/UserCommands.cc:690]: This could probably be gzdbg

File: src/systems/user_commands/UserCommands.cc [hunk 4]
```diff
@@ -859,7 +871,6 @@ void UserCommandsPrivate::OnCmdLight(const msgs::Light &_msg)
   }
 }
 
-
```
Comment (mjcarroll) [inline src/systems/user_commands/UserCommands.cc:862]: nit: unnecessary whitespace change

File: src/systems/user_commands/UserCommands.cc [hunk 5]
```diff
@@ -1362,79 +1373,142 @@ LightCommand::LightCommand(msgs::Light *_msg,
 //////////////////////////////////////////////////
 bool LightCommand::Execute()
 {
-  auto lightMsg = dynamic_cast<const msgs::Light *>(this->msg);
+  auto lightMsg = dynamic_cast<msgs::Light *>(this->msg);
```
Comment (mjcarroll) [inline src/systems/user_commands/UserCommands.cc:1376]: Is there a reason to remove the const?  You shouldn't be mutating the message, correct?

Comment (bperseghetti) [reply src/systems/user_commands/UserCommands.cc:1376]: Have to set the changes to the entity ID for each match found in the for loop.

Comment (bperseghetti) [reply src/systems/user_commands/UserCommands.cc:1376]: https://github.com/gazebosim/gz-sim/blob/149f9a28b9b8e026a90f8e65f43a53331b5fe397/src/systems/user_commands/UserCommands.cc#L1396

File: src/systems/user_commands/UserCommands.cc (hunk 4 above)
Comment (bperseghetti) [reply src/systems/user_commands/UserCommands.cc:862]: My eyes didn't like the lack of uniformity by leaving the extra space. Haha
image

Comment (bperseghetti) [reply src/systems/user_commands/UserCommands.cc:862]: Feel free to re-open if change needed.

**Review by mjcarroll — APPROVED** [review]
